### PR TITLE
PR - Code Review - Cataloger app with working tests and linters

### PR DESCRIPTION
**Pull request by wickathou**
For this PR, a complete cataloging app has been created, following proper collaborative strategies and following gitflow.

**Stack:**
- PostgreSQL
- Ruby

**Features:**
- [x] Complete functional interface to create and list items
- [x] Categories for items based on classes Book, MusicAlbum and Game
- [x] Specific properties such as Labels, Genres and Authos can be assigned and created dynamically for items
- [x] Complete database schema of the app

[**Video walkthrough**](https://drive.google.com/file/d/1SqGdtflpsnJATdV6KLpV0DLCJ28ngdBt/view?usp=share_link)

### CODE REVIEW

File: lib/save_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+#         last_played_date: instance.last_played_date,
+#         items: instance.items.map(&:id)
+#       }
+#     end
+#   end
+
+#   def save_json
+#     File.write('./data/authors.json', to_json)
+#   end
+# end
```
Comment (AshakaE) [inline lib/save_decorator.rb]: - [x] Kindly remove all comments in your code, this is to ensure best practices and clean code

File: README.md [hunk 2]
```diff
@@ -1,2 +1,146 @@
...
+- **Add books, music albums and games**
+- **List all books, labels, music albums, genres, games & authors**
+- **Data preserved in json file**
+- **Schema file with tables of the classes in the app**
+
+<!-- video DEMO -->
+
+## 🚀 Video Demo <a name="video-demo"></a>
+
+- [Video Presentation](https://drive.google.com/file/d/1qlVujgklgt2I1tAiO1a93D11yZplsaB0/view?usp=sharing)
```
Comment (AshakaE) [inline README.md]: Your presentation is a bit long and it is over the maximum time allowed

- [x] Kindly record a new video and keep it under the maximum length of 5 mins

**Review by AshakaE — CHANGES_REQUESTED** [review]
Hi Team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Your test cases pass 👏 
- Pull request description ✅ 
- The use of Github flow ✅ 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: lib/item.rb [hunk 3]
```diff
@@ -0,0 +1,75 @@
+require_relative 'author'
+require 'securerandom'
+
+class Item
+  attr_accessor :author, :label, :genre
```
Comment (FrederickMih) [inline lib/item.rb:5]: - [Optional] author, label, and genre instance variables should only be read in this class. Thats  attr_reader :author, :label, :genre

File: lib/app.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  end
+
+  def app_loop
+    loop do
+      interface_menu
+      option = gets.chomp
+      if option == '10'
+        app_exit
+        break
+      end
```
Comment (FrederickMih) [inline lib/app.rb:49]: - [Optional] The app exit with some errors as can be seen in the screenshot below: Let's kindly fix :wrench: the issue.
Screenshot from 2023-05-05 23-02-49

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


## Highlights :green_circle: 

- PR is professional :+1: 
- Descriptive README.md file :+1: 


## Optional suggestions
**N/A**

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**